tests/raft_scenarios/4806

```diff
@@ -0,0 +1,20 @@
+## create_new_node should create a node in pending state,
+## A node with no connection with any other nodes
+## and not part of any configuration.
+create_new_node,0
+
+periodic_all,500
+dispatch_all
+state_all
+
+## The node should *not* be a primary, but a
+## follower.
+assert_is_candidate,0
```

Comment:
This is contradictory with the logic added on line 1886 that prevents a node from becoming candidate if there is no configuration.

Reply:
4806 is a "negative test" that should fail with the given change in raft.h. If we all agree that the scenario in 4806 should not be possible and the code change is correct, I will remove 4806.

Reply:
By the way, this test exposes a limitation of the existing assertions provided in driver.h; it would be beneficial if it also included assert_is_not_* or a more general assert_is_state that accepts arguments.

Reply:
Yes, we need to revisit that wrapper. It was trying to be generic and encapsulate state at a time when the prevailing thought was that there would be several consensus implementations that could easily be swapped in. This no longer the case, and we'd rather have a more precise way to talk about the existing implementation instead.

Reply:
I'd expect https://github.com/microsoft/CCF/issues/5057 to take over a significant part of scenario *validation*, and for the wrapper to be "reduced" to being the stimulus.